src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -75,27 +75,25 @@ internal static Schema GetModelSchema(IExceptionContext ectx, TFGraph graph, str
         /// of kind <see cref="OpType"/>, indicating the operation type of the node, and if that node has inputs in the graph,
         /// it contains metadata of kind <see cref="InputOps"/>, indicating the names of the input nodes.
         /// </summary>
-        /// <param name="ectx">An <see cref="IExceptionContext"/>.</param>
-        /// <param name="modelFile">The name of the file containing the TensorFlow model. Currently only frozen model
-        /// format is supported.</param>
-        public static Schema GetModelSchema(IExceptionContext ectx, string modelFile)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="modelPath">Model to load.</param>
+        public static Schema GetModelSchema(IHostEnvironment env, string modelPath)
         {
-            var bytes = File.ReadAllBytes(modelFile);
-            var session = LoadTFSession(ectx, bytes, modelFile);
-            return GetModelSchema(ectx, session.Graph);
+            var model = LoadTensorFlowModel(env, modelPath);
+            return GetModelSchema(env, model.Session.Graph);
         }
 
         /// <summary>
         /// This is a convenience method for iterating over the nodes of a TensorFlow model graph. It
-        /// iterates over the columns of the <see cref="ISchema"/> returned by <see cref="GetModelSchema(IExceptionContext, string)"/>,
+        /// iterates over the columns of the <see cref="ISchema"/> returned by <see cref="GetModelSchema(IHostEnvironment, string)"/>,
         /// and for each one it returns a tuple containing the name, operation type, column type and an array of input node names.
         /// This method is convenient for filtering nodes based on certain criteria, for example, by the operation type.
         /// </summary>
-        /// <param name="modelFile"></param>
+        /// <param name="modelPath">Model to load.</param>
         /// <returns></returns>
-        public static IEnumerable<(string, string, ColumnType, string[])> GetModelNodes(string modelFile)
+        public static IEnumerable<(string, string, ColumnType, string[])> GetModelNodes(string modelPath)
         {
-            var schema = GetModelSchema(null, modelFile);
+            var schema = GetModelSchema(new MLContext(), modelPath);
```

Comment:
This is not correct. If you need an `IHostEnvironment`, you should be taking one as a parameter. It is wrong to create new environment (`MLContext`) out of thin air.

/cc @TomFinley

Reply:
This is only used as an IExceptionContext, so if we remove the check that env!=null on line 325, we can pass null.

---
In reply to: [247276711](https://github.com/dotnet/machinelearning/pull/2112#discussion_r247276711) [](ancestors = 247276711)